src/main/webapp/go.js

```diff
@@ -150,30 +185,39 @@ function toggleHintButton(hide) { //eslint-disable-line
  * Disable lint check because getHint() is called from go.html.
  */
 function getHint() { //eslint-disable-line
-  if (destIndex >= 0) {
-    const hintArr = huntArr[destIndex].hints;
+  if (hunt.getDestIndex() >= 0) {
+    const hintArr = hunt.getCurHints();
+    const hintIndex = hunt.getHintIndex();
     if (hintIndex < hintArr.length) {
       updateMessage(HINT_DISPLAY, hintArr[hintIndex]);
-      hintIndex++;
+      hunt.setHintIndex(hintIndex + 1);
     }
   }
 }
 
 /**
  * Determines whether the user entered the correct destination, and
  * adjusts the display accordingly.
+ * Disable lint check because handleDestinationAnswer() is called
+ * from go.html.
  */
-function handleDestinationAnswer() {
-  fetch(GUESS_URL).then((response) => response.json()).then((guess) => {
-    if (guess === huntArr[destIndex].name) {
-      toggleProceedButton(/* hide = */ false);
-      toggleHintButton(/* hide = */ true);
+function handleDestinationAnswer() { //eslint-disable-line
+  if (hunt.getDestIndex() == -1) { // User has not yet started the hunt.
+    return;
+  }
+  const userGuess = document.getElementById(GUESS_DISPLAY).value;
+  const queryStr = GUESS_URL + '?user-input=' + userGuess +
+      '&answer=' + hunt.getCurName();
+  fetch(queryStr).then((response) => response.json()).then((result) => {
```

Comment:
You can call this variable whatever you want.  Suggest calling it something like correctGuess so it's clear that a true value means they guessed correctly.
```suggestion
  fetch(queryStr).then((response) => response.json()).then((correctGuess) => {
```